### DISCUSSION TITLE
FlxSprite ImageData/bitmap to shaders via uniform

### DISCUSSION THREAD

**Discussion by bogdan-ov** [HaxeFlixel/flixel#2692] [Q&A]
I want to make my own mask shader for the sprite (I know about FlxSpriteUtil, but that's not what I need)
I tried to do it like this, but got an error:
```haxe
class MyMaskShader extends FlxShader {
    @:glFragmentSource("
        #pragma header

        uniform sampler2 uMask;

        void main() {
            vec4 color = flixel_texture2D(bitmap, openfl_TextureCoordv);
            vec4 mask = flixel_texture2D(uMask, openfl_TextureCoordv);

            gl_FragColor = vec4(color.r, color.g, color.b, color.a * uMask.a);
        }
    ")

    public function new() {
        super();
    }

    public function update(maskSprite:FlxSprite) {
        uMask.input = maskSprite.graphic.bitmap;
    }
}
```
Errors:
![image](https://user-images.githubusercontent.com/60233692/205485880-325b40fa-699a-4c44-a5b7-d98489dcaeae.png)
![image](https://user-images.githubusercontent.com/60233692/205485885-54c93bba-190a-46e4-a76f-82bd822704fc.png)

**Thank you in advance!**

**Comment by Geokureli** [comment]
the error should contain info about what went wrong, and what actual gl code was generated from your haxe code. I'm not sure what the error is based on these screenshots alone
